src/Network/Ricochet/Types.hs

```diff
@@ -11,41 +19,45 @@ import           System.IO       (Handle ())
 
 -- | Low level representation of a ricochet packet
 data Packet = MkPacket
-  { _pSize       :: Word16
-  , _pChannelID  :: Word16
-  , _pPacketData :: ByteString
+  { _pSize       :: Word16     -- ^ The size of the whole packet
+  , _pChannelID  :: Word16     -- ^ The channel the packet should be sent on
+  , _pPacketData :: ByteString -- ^ The actual packet payload
   } deriving (Show)
 
--- | Creates a packet with appropiate size from a Channel and a ByteString
+-- | Creates a packet with appropriate size from a Channel and a ByteString
 makePacket :: Word16     -- ^ ID of the channel the packet should be sent on
            -> ByteString -- ^ The ByteString to be sent
            -> Packet     -- ^ Returns a sendable packet
 makePacket chan bs = MkPacket (4 + fromIntegral (B.length bs)) chan bs
 
+-- | The role of a peer in a Connection
+data ConnectionRole = Client | Server deriving (Eq, Show)
+
 -- | Representation of a connection between two ricochet users
 -- it consists of:
 --  - open channels
 --  - wether our ricochet instance is the client
 -- Equality is defined by equality of the socket
```

Comment:
I still think this should be moved to the instance definition.  Did you try out if it works?